kafka_scanner/msg_processor_handlers.py

```diff
@@ -20,11 +20,23 @@ def retry_on_exception(exception):
 
 
 class MsgProcessorHandlers(object):
-    def __init__(self, encoding=None):
-        self.decompress_fun = zlib.decompress
+    def __init__(self, encoding=None, decompress=None, msgformat=None):
+        if decompress:
+            import zlib
+            self.decompress_fun = zlib.decompress
+        else:
+            self.decompress_fun = self.dummy_decompress
         self.consumer = None
         self.__next_messages = 0
         self.__encoding = encoding
+        self.deserialize_fun = self.set_deserializer(msgformat)
+
+    def set_deserializer(self, msg_format):
```

Comment:
in the way you are using this method, seems better a name like get_deserializer(). The method doesn't really set anything. Only returns an object than it is set in previous line.

Reply:
good point, naming is important